var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -511,21 +511,11 @@ def install(self, spec, prefix):
             make('check PETSC_ARCH="" PETSC_DIR={0}'.format(self.prefix),
                  parallel=False)
 
-    def setup_build_environment(self, env):
-        # configure fails if these env vars are set outside of Spack
-        env.unset('PETSC_DIR')
-        env.unset('PETSC_ARCH')
-
     def setup_run_environment(self, env):
         # Set PETSC_DIR in the module file
         env.set('PETSC_DIR', self.prefix)
         env.unset('PETSC_ARCH')
 
-    def setup_dependent_build_environment(self, env, dependent_spec):
-        # Set up PETSC_DIR for everyone using PETSc package
-        env.set('PETSC_DIR', self.prefix)
-        env.unset('PETSC_ARCH')
-
```

Comment:
same here.

Reply:
What is the `dependent_spec` for? I don't see it being used.

Reply:
This env is setup for the build of dependent packages like slepc

Reply:
Ok, but is `dependent_spec` needed as a parameter?

Reply:
Check `git grep 'def setup_dependent_build_environment'`

The usage here is same as usage from  many other packages.

What issue did you encounter that required this code removal?

Reply:
I see. I didn't run into an issue with the code. I removed the code because I wasn't sure it was necessary. I'll re-add it.